torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -348,7 +373,21 @@ class ShapePropagator {
     setUnshapedType(cat_node);
   }
 
+  bool mayAliasResizedSet(at::ArrayRef<Value*> vs) {
+    return std::any_of(vs.begin(), vs.end(), [&](Value* v) {
+      return aliasDb_.mayAlias({v}, resized_alias_set);
+    });
+  }
+
   void PropagateShapeOnNode(Node* node, bool insert_expands = true) {
+    // Certain ops like resize_ change the input tensors size. Because our
+    // is flow invariant, we set any Tensor that can alias a resized Tensor
+    // to the base Tensor Type without size information.
+    if (mayAliasResizedSet(node->inputs()) ||
+        mayAliasResizedSet(node->outputs())) {
```

Comment:
The check on outputs is not necessary. Either it aliases one of the inputs (in which case you've already checked it) or it's a new tensor.